### DISCUSSION TITLE
Project: Reducing Ethereum gas costs for crypto exchanges using layer 2 platforms.

### DISCUSSION THREAD

**Discussion by AlirezaEthDev** [ethereum-optimism/developers#249] [General]
### High level description of project.

As I researched on reducing Ethereum gas costs for crypto exchanges using layer 2 platforms, the use of these platforms is beneficial for an crypto exchange when the users of that exchange, after bridging their ethers to layer 2, execute transactions that their "from" and "to" must be in layer 2, otherwise, if "from" or "to" is in the Ethereum mainnet, then that transaction must be sent from Ethereum to layer 2 or from layer 2 to Ethereum, and in this case, the Ethereum gas fee must be paid. Is my conclusion correct? If yes, is there a better way to reduce gas costs for transactions that their "from" or "to" are in Ethereum mainnet? In general, what is the best solution for reducing Ethereum gas costs on an crypto exchange with low user scale?

### Label Checks

- [X] I have added the team label for this project (CH, TH, Contributions).
- [X] I have added the timeline for this project to the issue.

**Comment by imtipi** [comment]
https://docs.optimism.io/builders/cex-wallet-developers/cex-support#transaction-fees
you still need to pay L1 data fee for transaction on op mainnet
and do you mean you want to reduce gas fees while processing user withdrawal requests?it isn't related to the layer1 and layer2 tbh...

**Reply by AlirezaEthDev** [reply]
Yes, exactly. I want to reduce gas fees while processing user withdrawal requests from the crypto exchange's wallets.

**Comment by hash1go** [comment]
If users want to use L2 and the exchange only supports L1, they will have to pay additional transaction fees.
Also, since the [standard bridge](https://app.optimism.io/bridge/deposit) on OP Mainnet has high transaction costs, users can opt for cheaper [third-party bridges](https://www.optimism.io/apps/bridges). This is a viable option for small transactions as it can result in slippage.

**Reply by AlirezaEthDev** [reply]
Yes, the standard bridge protocol of OP requires high transaction costs. Also, the transaction costs of third-party bridges have costs more than Ethereum native ones.
